addons/mrp/models/mrp_production.py

```diff
@@ -849,19 +849,14 @@ def _can_produce_serial_number(self, sn=None):
         self.ensure_one()
         sn = sn or self.lot_producing_id
         if self.product_id.tracking == 'serial' and sn:
-            message, dummy = self.env['stock.quant'].sudo()._check_serial_number(self.product_id, sn, self.company_id)
+            message, recommended_location = self.env['stock.quant'].sudo()._check_serial_number(self.product_id, sn, self.company_id)
+            is_reproducing = self.lot_producing_id and self.lot_producing_id.id in self.move_raw_ids.lot_ids.ids
+            if is_reproducing and not recommended_location:
```

Comment:
Since we don't give any `source_location_id` to `_check_serial_number`, how could `recommended_location` be not null?